pkg/apiserver/request/requestinfo.go

```diff
@@ -193,6 +194,12 @@ func (r *RequestInfoFactory) NewRequestInfo(req *http.Request) (*RequestInfo, er
 		}
 	}
 
+	// URL forms: /api/v1/watch/namespaces?labelSelector=kubesphere.io/workspace=system-workspace
+	workspaceSelector := req.URL.Query().Get("labelSelector")
+	if groups := regexp.MustCompile(`^kubesphere.io/workspace=(\S+)$`).FindStringSubmatch(workspaceSelector); len(groups) == 2 {
```

Comment:
regex expression can be slow, and request info is in a critical code path, I'm not confident to put a regex expression here. And also, workspaceSelector can be empty, if then there is no need to do regex match

Reply:
fixed,  use  `strings.HasPrefix` instead of `regex expression`.